src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsLoggingBuilderExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Logging.ApplicationInsights;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.Configuration;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Extensions for ApplicationInsights configuration on an <see cref="ILoggingBuilder"/>. 
+    /// </summary>
+    public static class ApplicationInsightsLoggingBuilderExtensions
+    {
+        /// <summary>
+        /// Registers Application Insights and <see cref="ApplicationInsightsLoggerProvider"/> with an <see cref="ILoggingBuilder"/>.
+        /// </summary>        
+        public static ILoggingBuilder AddApplicationInsights(
+            this ILoggingBuilder builder)
+        {
+            return builder.AddApplicationInsights(null);
+        }
+
+        /// <summary>
+        /// Registers Application Insights and <see cref="ApplicationInsightsLoggerProvider"/> with an <see cref="ILoggingBuilder"/>.
+        /// </summary>        
+        public static ILoggingBuilder AddApplicationInsights(
+            this ILoggingBuilder builder,
+            Action<ApplicationInsightsLoggerOptions> configure)
+        {
+            builder.AddConfiguration();
+            builder.Services.AddApplicationInsights(configure);
+
+            builder.Services.PostConfigure<LoggerFilterOptions>(o =>
+            {
+                // We want all logs to flow through the logger so they show up in QuickPulse.
+                // To do that, we'll hide all registered rules inside of this one. They will be re-populated
+                // and used by the FilteringTelemetryProcessor futher down the pipeline.
+
+                var matchingRules = o.Rules.Where(r =>
+                {
+                    return r.ProviderName == typeof(ApplicationInsightsLoggerProvider).FullName
+                        || r.ProviderName == ApplicationInsightsLoggerProvider.Alias;
+                }).ToArray();
+
+                foreach (var rule in matchingRules)
+                {
+                    o.Rules.Remove(rule);
+                }
+
+                o.Rules.Add(new ApplicationInsightsLoggerFilterRule(matchingRules.ToList()));
```

Comment:
Here's the new behavior -- we capture all rules applying to our logger and store them inside a custom rule for re-enabling later.